src/App.tsx

```diff
@@ -17,18 +58,38 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                setSelectStatus={setSelectStatus}
+              />
             </div>
 
-            <div className="block">
-              <Loader />
-              <TodoList />
-            </div>
+            {todos.length === 0
+              ? (
+                <Loader />
+              ) : (
+                <div className="block">
+                  <TodoList
+                    todos={visibleTodos}
+                    setIsTodoSelected={setIsTodoSelected}
+                    isTodoSelected={isTodoSelected}
+                    setSelectedTodo={setSelectedTodo}
+                    selectedTodo={selectedTodo}
+                  />
+                </div>
+              )}
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isTodoSelected && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          isTodoSelected={isTodoSelected}
+          setIsTodoSelected={setIsTodoSelected}
+        />
+      )}
```

Comment:
You can render your todo modal only when you have `selectedTodo`.
It's can help you to remove unnecessary `isTodoSelected` from your code

```suggestion
      {selectedTodo && (
        <TodoModal
          selectedTodo={selectedTodo}
          setIsTodoSelected={setIsTodoSelected}
        />
      )}
```